list.js

```diff
@@ -0,0 +1,18 @@
+const arrayToList = (arr) => {
+    let list;
+    for (let i = 0; i < arr.length; i++) {
+        list = {
+            value: arr[i],
+            rest: list
+        }
+    }
+    return list
+}
+
+const listToArray = (list) => {
+    let arr = []
+    for (let num = list; num; num = num.rest) {
+        arr.push(num.value)
```

Comment:
Again, It's fine. (though I don't really like these reassignments in loops, it's hard to follow) 
Here is how this could be implemented using recursion in a pure way, just for the reference.
```javascript
const listToArray = ({ value, rest }, acc = []) => {
	const cumulated  = [...acc, value]
	return rest.length === 0 ? cumulated : listToArray(rest, cumulated)
}
```